CMakeLists.txt

```diff
@@ -0,0 +1,74 @@
+cmake_minimum_required(VERSION 3.14)
+
+project(pennylane_lightning)
+
+# Read and set pennylane_lightning version
+function(set_pennylane_lightning_version VERSION_FILE_PATH)
+	file(STRINGS ${VERSION_FILE_PATH} VERSION_FILE_STR)
+	foreach (LINE IN LISTS VERSION_FILE_STR)
+	if("${LINE}" MATCHES "__version__.*")
+		set(VERSION_LINE_STR "${LINE}")
+	endif()
+	endforeach()
+
+	string(REGEX REPLACE "__version__ = \"(.*)\"" "\\1" VERSION_STRING ${VERSION_LINE_STR})
+	set(VERSION_STRING ${VERSION_STRING} PARENT_SCOPE)
+endfunction()
```

Comment:
Sure, I am fine with this --- as you said earlier, doing file operations with CMake is never fun, but this approach is understandable.